src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -76,43 +85,27 @@ public void ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled(bool
             else
             {
                 fileToOpen = GetTestFilePath() + ".txt";
-                File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled)}");
+                File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecute_OnUnix_SuccessWhenProgramInstalled)}");
             }
-
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || _allowedProgramsToRun.Any(program => IsProgramInstalled(program)))
             {
-                string[] allowedProgramsToRun = { "xdg-open", "gnome-open", "kfmclient" };
-                foreach (var program in allowedProgramsToRun)
+                using (var px = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen }))
                 {
-                    if (IsProgramInstalled(program))
+                    Assert.NotNull(px); // on Windows px is null for some reason
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                     {
-                        programToOpen = program;
-                        break;
+                        // Assert.Equal(programToOpenWith, px.ProcessName); // on OSX, process name is dotnet for some reason. Refer to #23972
+                        Console.WriteLine($"{nameof(ProcessStart_UseShellExecute_OnUnix_SuccessWhenProgramInstalled)}(isFolder: {isFolder}), ProcessName: {px.ProcessName}");
```

Comment:
We shouldn't be doing Console.WriteLines on success paths in tests.

Reply:
@stephentoub I am adding the Console.WriteLine in lieu of the assert statement on line 98 which is commented out. I included the issue# as well so it gets tracked

Reply:
I'm not sure how that makes it better. It now spams every run with this info.

Reply:
The only purpose of the CWL is to be able to occasionally verify that at least on some of our test images, we are successfully finding and launching xdg-open. I think that is accomplished by the "..None of the following programs were installed on this machine.." higher up. That one line seems little bother since it's outerloop.
This one here doesn't really add any information.

Reply:
sure. will take note to remove this on next PR.

Reply:
Thanks